### PR TITLE
RDCC-2780: Upgrading Spring Version to 2.4.5

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-2780

### Change description ###

Upgrading Spring Version to 2.4.5 & other dependency versions

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileCreator.java [hunk 1]
```diff
@@ -155,16 +153,16 @@ private UserProfile handleDuplicateUser(UserProfileCreationData profileData,
...
         UserProfile userProfile = null;
         IdamRolesInfo idamRolesInfo;
         ResponseEntity<Object> responseEntity = idamRegistrationInfo.getResponse();
 
         if (responseEntity != null) {
             //get userId from location header
-            userIdUri = idamRegistrationInfo.getResponse().getHeaders().getLocation();
-            userId = userIdUri != null ? userIdUri.toString().substring(sidamGetUri.length()) : null;
+            locationHeader = idamRegistrationInfo.getResponse().getHeaders().getFirst("location");
+            userId = locationHeader != null ? locationHeader.substring(sidamGetUri.length()) : null;
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileCreator.java:165]: replace with utility method nonNull()

**Review by abhijitdiwan — DISMISSED** [review]

**Review by kavithagade12 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileCreator.java (hunk 1 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileCreator.java]: Change made here: https://github.com/hmcts/rd-user-profile-api/pull/688/commits/2ad83d1357bad1b9a5afb64f9a82499f0a1a6e8d

**Review by kavithagade12 — DISMISSED** [review]

**Review by shreedhar-hmcts — DISMISSED** [review]
Please test toggle OFF Delete API and 409 scenario

**Review by shreedhar-hmcts — APPROVED** [review]

**Review by shreedhar-hmcts — APPROVED** [review]

**Review by shreedhar-hmcts — DISMISSED** [review]

**Review by SujeethKumarAri — APPROVED** [review]

**Comment by AdilOoze** [comment]
Closing due to Nightly PR Issue blocking merge.

**Pull request closed without merging**